services/src/main/java/org/keycloak/crypto/Aes128GcmEncryptor.java

```diff
@@ -0,0 +1,39 @@
+package org.keycloak.crypto;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.UUID;
+
+public class Aes128GcmEncryptor {
+    private static final int IV_SIZE_BYTE = 12;
+    private static final int AUTH_TAG_SIZE_BYTE = 16;
+    private static final String ALGORITHM = "AES/GCM/NoPadding";
+    private static final byte[] IV = new byte[IV_SIZE_BYTE];
+
+    private Aes128GcmEncryptor() {
+    }
+
+    public static byte[] encrypt(String secret, byte[] bytes) throws GeneralSecurityException {
+        return getCipher(Cipher.ENCRYPT_MODE, secret).doFinal(bytes);
+    }
+
+    public static byte[] decrypt(String secret, byte[] encryptedBytes) throws GeneralSecurityException {
+        return getCipher(Cipher.DECRYPT_MODE, secret).doFinal(encryptedBytes);
+    }
+
+    private static Cipher getCipher(int mode, String secret) throws GeneralSecurityException {
+        UUID uuid = UUID.fromString(secret);
+        byte[] bytes = ByteBuffer.wrap(new byte[AUTH_TAG_SIZE_BYTE])
+                               .putLong(uuid.getMostSignificantBits())
+                               .putLong(uuid.getLeastSignificantBits())
+                               .array();
+        SecretKeySpec secretKey = new SecretKeySpec(bytes, "AES");
```

Comment:
AFAIK, keycloak tends to use BC (BouncyCastle) as security provider, not SunJCE.

https://github.com/tnorimat/keycloak/blob/be4780243b702e9b7e8aa236bbde1312b93ebfea/core/src/main/java/org/keycloak/jose/jwe/enc/AesGcmEncryptionProvider.java#L91

https://github.com/tnorimat/keycloak/blob/be4780243b702e9b7e8aa236bbde1312b93ebfea/core/src/main/java/org/keycloak/jose/jwe/enc/AesGcmEncryptionProvider.java#L101

Reply:
updated to use BouncyCastle